config/v1alpha1/types_image_policy.go

```diff
@@ -0,0 +1,215 @@
+type PolicyRootOfTrust struct {
+	// policyType serves as the union's discriminator. Users are required to assign a value to this field, choosing one of the policy types that define the root of trust.
+	// "PublicKey" indicates that the policy relies on a PGP publicKey and may optionally use a Rekor verification.
+	// "FulcioCAWithRekor" indicates that the policy is based on the Fulcio certification and incorporates a Rekor verification.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	PolicyType PolicyType `json:"policyType"`
+	// PublicKey defines the root of trust based on a PGP public key.
+	// +optional
+	PublicKey *PublicKey `json:"publicKey,omitempty"`
+	// FulcioCAWithRekor defines the root of trust based on the Fulcio certificate and the Rekor public key.
+	// For more information about Fulcio and Rekor, please refer to the document at:
+	// https://github.com/sigstore/fulcio and https://github.com/sigstore/rekor
+	// +optional
+	FulcioCAWithRekor *FulcioCAWithRekor `json:"fulcioCAWithRekor,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=PublicKey;FulcioCAWithRekor
+type PolicyType string
+
+const (
+	PublicKeyRootOfTrust         PolicyType = "PublicKey"
+	FulcioCAWithRekorRootOfTrust PolicyType = "FulcioCAWithRekor"
+)
+
+// PublicKey defines the root of trust based on a PGP public key.
+type PublicKey struct {
+	// keyData contains inline base64 encoded data of the public key.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=8192
+	KeyData string `json:"keyData"`
+	// rekorKeyData contains inline base64 data of the Rekor public key.
+	// +optional
+	// +kubebuilder:validation:MaxLength=8192
+	RekorKeyData string `json:"rekorKeyData,omitempty"`
+}
+
+// FulcioCAWithRekor defines the root of trust based on the Fulcio certificate and the Rekor public key.
+type FulcioCAWithRekor struct {
+	// fulcioCAData contains inline base64 data for the fulcio CA certificate.
+	// +kubebuilder:validation:Required
+	FulcioCAData string `json:"fulcioCAData"`
+	// rekorKeyData contains inline base64 data of the Rekor public key.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=8192
+	RekorKeyData string `json:"rekorKeyData"`
+	// fulcioSubject specifies OIDC issuer and the email of the Fulcio authentication configuration.
+	// Required if rootOfTrust is based on Fulcio.
+	// +kubebuilder:validation:Required
+	FulcioSubject PolicyFulcioSubject `json:"fulcioSubject,omitempty"`
+}
+
+// PolicyFulcioSubject defines the OIDC issuer and the email of the Fulcio authentication configuration.
+type PolicyFulcioSubject struct {
+	// oidcIssuer contains the expected OIDC issuer.
+	// Example: "https://expected.OIDC.issuer/"
+	// +kubebuilder:validation:Required
+	OIDCIssuer string `json:"oidcIssuer"`
+	// signedEmail holds the email address the the certificate is issued for.
```

Comment:
Which certificate are you talking about here?

Reply:
fixed by specifying it's the Fulcio certificate